test/extensions/stats_sinks/metrics_service/grpc_metrics_service_impl_test.cc

```diff
@@ -186,17 +186,36 @@ TEST_F(MetricsServiceSinkTest, ReportCountersValues) {
 
 // Test that verifies counters are reported as the delta between flushes when configured to do so.
 TEST_F(MetricsServiceSinkTest, ReportCountersAsDeltas) {
-  MetricsServiceSink<envoy::service::metrics::v3::StreamMetricsMessage,
-                     envoy::service::metrics::v3::StreamMetricsResponse>
-      sink(streamer_, true, false);
-
   addCounterToSnapshot("test_counter", 1, 100);
+  counter_storage_.back()->setTagExtractedName("tag-counter-name");
+  counter_storage_.back()->setTags({{"a", "b"}});
 
-  EXPECT_CALL(*streamer_, send(_)).WillOnce(Invoke([](MetricsPtr&& metrics) {
-    EXPECT_EQ(1, metrics->size());
-    EXPECT_EQ(1, (*metrics)[0].metric(0).counter().value());
-  }));
-  sink.flush(snapshot_);
+  {
+    MetricsServiceSink<envoy::service::metrics::v3::StreamMetricsMessage,
+                       envoy::service::metrics::v3::StreamMetricsResponse>
+        sink(streamer_, true, false);
+
+    EXPECT_CALL(*streamer_, send(_)).WillOnce(Invoke([](MetricsPtr&& metrics) {
+      EXPECT_EQ(1, metrics->size());
+      EXPECT_EQ(1, (*metrics)[0].metric(0).counter().value());
```

Comment:
make a temp for *metrics)[0].metric(0).counter() and also check its name?

Also let's check here that there is no label.